src/ir_Technibel.cpp

```diff
@@ -0,0 +1,416 @@
+  uint8_t mode = GETBITS64(remote_state, kTechnibelAcModeOffset,
+                            kTechnibelAcModeSize);
+  return reverseBits(mode, kTechnibelAcModeSize);
+}
+
+void IRTechnibelAc::setMode(const uint8_t mode) {
+  switch (mode) {
+    case kTechnibelAcHeat:
+    case kTechnibelAcFan:
+    case kTechnibelAcDry:
+    case kTechnibelAcCool:
+      break;
+    default:
+      this->setMode(kTechnibelAcCool);
+      return;
+  }
+  uint8_t mode_reversed = reverseBits(mode, kTechnibelAcModeSize);
+  setBits(&remote_state, kTechnibelAcModeOffset, kTechnibelAcModeSize,
+          mode_reversed);
+  setFan(getFan());  // Re-force any fan speed constraints.
+  // Restore previous temp settings for cool mode.
+  setTemp(_saved_temp, _saved_temp_units);
+}
+
+// Convert a standard A/C mode into its native mode.
+uint8_t IRTechnibelAc::convertMode(const stdAc::opmode_t mode) {
+  switch (mode) {
+    case stdAc::opmode_t::kCool:
+      return kTechnibelAcCool;
+    case stdAc::opmode_t::kHeat:
+      return kTechnibelAcHeat;
+    case stdAc::opmode_t::kDry:
+      return kTechnibelAcDry;
+    case stdAc::opmode_t::kFan:
+      return kTechnibelAcFan;
+    default:
+      return kTechnibelAcCool;
+  }
+}
+
+// Convert a native mode to it's common equivalent.
+stdAc::opmode_t IRTechnibelAc::toCommonMode(const uint8_t mode) {
+  switch (mode) {
+    case kTechnibelAcCool:  return stdAc::opmode_t::kCool;
+    case kTechnibelAcHeat:  return stdAc::opmode_t::kHeat;
+    case kTechnibelAcDry:   return stdAc::opmode_t::kDry;
+    case kTechnibelAcFan:   return stdAc::opmode_t::kFan;
+    default:                return stdAc::opmode_t::kAuto;
+  }
+}
+
+void IRTechnibelAc::setSwing(const bool on) {
+  setBit(&remote_state, kTechnibelAcSwingBit, on);
+}
+
+bool IRTechnibelAc::getSwing(void) {
+  return GETBIT64(remote_state, kTechnibelAcSwingBit);
+}
+
+// Convert a standard A/C swing into its native swing.
+bool IRTechnibelAc::convertSwing(const stdAc::swingv_t swing) {
+  switch (swing) {
+    case stdAc::swingv_t::kOff:
+      return false;
+    case stdAc::swingv_t::kAuto:
+      return true;
+    default:
+      return true;
+  }
+}
+
+// Convert a native swing to it's common equivalent.
+stdAc::swingv_t IRTechnibelAc::toCommonSwing(const bool swing) {
+  if (swing)
+    return stdAc::swingv_t::kAuto;
+  else
+    return stdAc::swingv_t::kOff;
+}
+
+void IRTechnibelAc::setSleep(const bool on) {
+  setBit(&remote_state, kTechnibelAcSleepBit, on);
+}
+
+bool IRTechnibelAc::getSleep(void) {
+  return GETBIT64(remote_state, kTechnibelAcSleepBit);
+}
+
+void IRTechnibelAc::setTimerEnabled(const bool on) {
+  setBit(&remote_state, kTechnibelAcTimerEnableBit, on);
+}
+
+bool IRTechnibelAc::getTimerEnabled(void) {
+  return GETBIT64(remote_state, kTechnibelAcTimerEnableBit);
+}
+
+// Set the timer to shutdown the device in nr of minutes.
+// Args:
+//   nr_of_hours: Total nr of hours to wait before shutting down the device.
+//               (Max 24 hrs)
```

Comment:
If possible, please use `nr_of_minutes` to keep it consistent with the rest of the protocols.
i.e. Just do a `/ 60` in the function procedure.